src/cata_tiles.cpp

```diff
@@ -179,66 +193,212 @@ void cata_tiles::get_tile_information(std::string config_path, std::string &json
     }
 }
 
+inline pixel cata_tiles::get_pixel_color(SDL_Surface *surf, int x, int y, int w)
+{
+    pixel pix;
+    const auto pixelarray = reinterpret_cast<unsigned char *>(surf->pixels);
```

Comment:
I'll have to poke at the SDL documentation a bit, but usually it provides a cleaner interface than this to manipulate data, there are probably some setter/getter macros or something that would encapsulate all this.